bot.lua

```diff
@@ -5267,52 +5285,62 @@ commands["gcmd"] = {
-								command = string.lower(command)
-
-								if commands[command] then
-									return sendError(message, "GCMD", "This command already exists and is not global.")
-								end
-								if globalCommands[command] and (globalCommands[command].author ~= message.author.id and not authIds[message.author.id]) then
-									return sendError(message, "GCMD", "This command already exists.", "Ask the owner, <@" .. globalCommands[command].author .. ">, for an edition.")
-								end
-
-								local cmd = getCommandTable(message, script, content, title, description)
-								if type(cmd) == "string" then
-									return sendError(message, "GCMD", cmd)
-								end
-
-								cmd.author = ((globalCommands[command] and globalCommands[command].author) or message.author.id)
-								cmd.auth = (authLevel == 0 and permissions.public or authLevel == 1 and permissions.is_dev or permissions.is_module)
-								if channelLevel == 1 then
-									cmd.category = message.channel.category.id
-								elseif channelLevel == 2 then
-									cmd.channel = message.channel.id
+							if allowDM then
+								allowDM = tonumber(allowDM)
+								if allowDM < 2 then
+									if command and #command > 1 and #command < 21 then
+										command = string.lower(command)
+
+										if commands[command] then
+											return sendError(message, "GCMD", "This command already exists and is not global.")
+										end
+										if globalCommands[command] and (globalCommands[command].author ~= message.author.id and not authIds[message.author.id]) then
+											return sendError(message, "GCMD", "This command already exists.", "Ask the owner, <@" .. globalCommands[command].author .. ">, for an edition.")
+										end
+
+										local cmd = getCommandTable(message, script, content, title, description)
+										if type(cmd) == "string" then
+											return sendError(message, "GCMD", cmd)
+										end
+
+										cmd.author = ((globalCommands[command] and globalCommands[command].author) or message.author.id)
+										cmd.auth = (authLevel == 0 and permissions.public or authLevel == 1 and permissions.is_dev or permissions.is_module)
+										cmd.dm = allowDM == 1
+										if channelLevel == 1 then
+											cmd.category = message.channel.category.id
+										elseif channelLevel == 2 then
+											cmd.channel = message.channel.id
+										end
+
+										globalCommands[command] = cmd
+
+										saveGlobalCommands()
+
+										message:reply({
+											embed = {
+												color = color.sys,
+												description = "Command `" .. command .. "` created successfully!",
+												footer = { text = "By " .. message.member.name }
+											}
+										})
+									else
+										sendError(message, "GCMD", "Invalid syntax.", syntax)
+									end
+								else
+									sendError(message, "GCMD", "Invalid level flag.", "The DM authorization level must be 0 (Disallowed) or 1 (Allowed).")
```

Comment:
could you add this for the other flags too, ty

Reply:
what do you mean?

Reply:
The send error message when there's a bad authlevel and channellevel, i think i never added it :P well, doesn't matter, i can do that later too

Reply:
I actually copy pasted the error message from the other flags.